x-pack/plugins/security_solution/common/detection_engine/schemas/types/only_false_allowed.ts

```diff
@@ -0,0 +1,33 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import * as t from 'io-ts';
+import { Either } from 'fp-ts/lib/Either';
+
+export type OnlyFalseAllowedC = t.Type<boolean, boolean, unknown>;
+
+/**
+ * Types the OnlyFalseAllowed as:
+ *   - If null or undefined, then a default false will be set
+ *   - If true is sent in then this will return an error
+ *   - If false is sent in then this will allow it only false
+ */
+export const OnlyFalseAllowed: OnlyFalseAllowedC = new t.Type<boolean, boolean, unknown>(
```

Comment:
I don't think you need to re-specify the generic arguments here, those should be inferred by the arguments (and enforced by `OnlyFalseAllowedC`)

Reply:
Yeah I can remove them in all the files, it was a pattern from io-ts I just modified from their examples.

It would be better to just use:

```ts
export type OnlyFalseAllowedC = typeof OnlyFalseAllowed;
```

And not re-type it twice everywhere. Will fix that.